synapse/events/snapshot.py

```diff
@@ -447,26 +425,27 @@ async def persist(self, event: EventBase) -> EventContext:
 
         # if the event isn't a state event the state group doesn't change
         if not self.state_delta_due_to_event:
-            state_group_after_event = self.state_group_before_event
+            self.state_group_after_event = self.state_group_before_event
 
         # otherwise if it is a state event we need to get a state group for it
         else:
-            state_group_after_event = await self._storage.state.store_state_group(
+            self.state_group_after_event = await self._storage.state.store_state_group(
                 event.event_id,
                 event.room_id,
                 prev_group=self.state_group_before_event,
                 delta_ids=self.state_delta_due_to_event,
                 current_state_ids=None,
             )
```

Comment:
Kind of seems like we always should have been setting these?

Reply:
(There's no action to take here, just weird that we didn't previously set these on `self`.)

Reply:
We were always tracking the value, just not storing it in self - I think I changed this just to have all the info in `self` so it could be passed to `_build_state_group_deltas` - either way the value is basically immediately used to create the `EventContext` on line 442 and is then never accessed again. But I do agree this way is more consistent and thus easier to follow.